### PR TITLE
New tool: git-safe-dependencies

### PR DESCRIPTION
**Pull request by naugtur**
A new tool to help people get their git dependencies safely.

It's all an elaborate linter.

A prompt review would mean I can release it as alpha and recommend to people online who are already concerned with my findings that this is addressing ;)

### CODE REVIEW

**Comment by naugtur** [comment]
@boneskull I'm pretty sure I messed up some of the setup for a new package

File: .release-please-manifest.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "packages/webpack": "0.7.0-beta.0",
   "packages/browserify": "18.1.2",
   "packages/core": "16.2.2",
+  "packages/git-safe-dependencies": "0.0.1",
```
Comment (leotm) [inline .release-please-manifest.json:7]: we planning to release it as alpha right? was expecting the word `alpha` in there 😛

similar to the `beta` in `"0.7.0-beta.0"`

File: packages/git-safe-dependencies/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# git-safe-dependencies
+
+> WARNING: This is an early version. The opinions are still forming and anything can change at any time. Test coverage is incomplete. Use at your own risk.
```
Comment (leotm) [inline packages/git-safe-dependencies/README.md]:
```suggestion
> WARNING: This is an alpha version. The opinions are still forming and anything can change at any time. Test coverage is incomplete. Use at your own risk.
```

File: packages/git-safe-dependencies/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+Validate your package.json and lockfile against an opinionated set of rules around using git dependencies that we at LavaMoat recommend.
```
Comment (leotm) [inline packages/git-safe-dependencies/README.md]:
```suggestion
Validate your `package.json` and lockfile (NPM, Yarn Classic or Yarn Berry) against an opinionated set of rules around using git dependencies that we at LavaMoat recommend.  
```

File: .release-please-manifest.json (hunk 1 above)
Comment (naugtur) [reply .release-please-manifest.json:7]: TBF, I'm not enjoying that way of publishing webpack plugin as release-please doesn't handle it for us. these are good for having a new major release be an alpha or RC. version starting with 0 is enough of an indication

File: packages/git-safe-dependencies/test/manual/yarn-classic-realistic/yarn.lock.ignore.json
Comment (leotm) [inline packages/git-safe-dependencies/test/manual/yarn-classic-realistic/yarn.lock.ignore.json:1]: do we remove what's here from when we scan, or is it an automatic lirantal/lockfile-lint thing

**Review by leotm — APPROVED** [review]

File: packages/git-safe-dependencies/test/manual/yarn-classic-realistic/yarn.lock.ignore.json
Comment (naugtur) [reply packages/git-safe-dependencies/test/manual/yarn-classic-realistic/yarn.lock.ignore.json:1]: the implementation of this is in cli.js

Comment (leotm) [reply packages/git-safe-dependencies/test/manual/yarn-classic-realistic/yarn.lock.ignore.json:1]: its used for the `--ignore` flag of the CLI (like the `--type` flag)

**Comment by boneskull** [comment]
@naugtur Well, it's definitely missing a TS config 🤣

**Pull request merged**